compiler/optimizer/VPHandlers.cpp

```diff
@@ -7328,6 +7339,14 @@ TR::Node *constrainIshr(OMR::ValuePropagation *vp, TR::Node *node)
       //lhs->decReferenceCount();
       //rhs->decReferenceCount();
       }
+
+   // Replace shift of constant zero with constant zero
+   if (lhs && lhs->asIntConst()
+       && lhs->asIntConst()->getInt() == 0
+       && performTransformation(vp->comp(), "%sRemoved redundant ishr node [%s]\n", OPT_DETAILS, node->getOpCode().getName())) {
+      return vp->replaceNode(node, node->getFirstChild(), vp->_curTree);
+   }
+
```

Comment:
I would suggest moving this transformation earlier in `constrainIshr`.  That way, if the left operand is zero, and the second operand is a constant, the IL can be simplified quickly rather than having to go through the extra work that's performed for the `if (rhs && rhs->asIntConst())` case above.

Same suggestion applies for `constrainLshr`, `constrainIushr` and `constrainLushr`.